Source/WTF/wtf/NativePromise.h

```diff
@@ -1513,19 +1525,23 @@ class NativePromiseProducer final : public ConvertibleToNativePromise {
     void setDispatchMode(PromiseDispatchMode dispatchMode, const Logger::LogSiteIdentifier& callSite) const
     {
         ASSERT(m_promise, "used after move");
-        m_promise->setDispatchMode(dispatchMode, callSite);
+        protectedPromise()->setDispatchMode(dispatchMode, callSite);
     }
 
     friend PromiseType;
     void assertIsDead() const
     {
         if (m_promise)
-            m_promise->assertIsDead();
+            protectedPromise()->assertIsDead();
     }
 
     // The Producer may be moved to resolve/reject the completion promise.
     // While we expect m_promise to never be null, it would cause a null dereference in the destructor if the destructor was called after a move.
     RefPtr<PromiseType> m_promise;
+    RefPtr<PromiseType> protectedPromise() const
```

Comment:
m_promise is never re-assigned, so we should mark the member variable const instead of making a protectedPromise function.